File: Math/Medium/LET_no1041_Robot Bounded In Circle.cpp
class Solution {
public:
    bool isRobotBounded(string instructions) {
        int i = 0, j = 0, d = 0;
        vector<vector<int>> moves = {{0, 1}, {1, 0}, {0, -1}, {-1, 0}};
        for (char c : instructions) {
            if (c == 'R')
                d = (d + 1) % 4;
            if (c == 'L')
                d = (d + 3) % 4;
            if (c == 'G') {
                i += moves[d][0];
                j += moves[d][1];
            }
        }
        return i == 0 && j == 0 || d != 0;
    }
};

File: Math/Medium/LET_no1041_Robot Bounded In Circle.py
class Solution:
    def isRobotBounded(self, instructions: str) -> bool:
        i, j, di, dj = 0, 0, 0, 1
        for c in instructions:
            if c == 'R': 
                di, dj = dj, -di
            if c == 'L':
                di, dj = -dj, di
            if c == 'G':
                i, j = i + di, j + dj
        return (i, j) == (0, 0) or (di, dj) != (0, 1)